skywalking/agent/__init__.py

```diff
@@ -32,18 +32,26 @@
 def __heartbeat():
     while not __finished.is_set():
         if connected():
-            __protocol.heartbeat()
-
-        __finished.wait(30 if connected() else 3)
+            try:
+                __protocol.heartbeat()
+                __finished.wait(3)
+            except Exception as e:
+                pass
+        else:
+            connect()
```

Comment:
There is another problem when the client **IS CONNECTING**, we should not `connect` again